### PR TITLE
fix(events): run the no-context background flush when the handler raises

### LINKED ISSUE #6982: Background handler that raises before entering `async with self` gets no compatibility delta flush
**Describe the bug**

Since the background-task delta-race fix, a background handler that returns without ever entering `async with self` still gets one "compatibility" delta flush under the state lock after it returns (refreshing uncached computed vars and preamble-dirty vars like `router_data`). But if such a handler *raises*, the exception propagates before the flush and no delta is emitted at all — the client is left without the refresh that every non-raising no-context background handler gets.

**To Reproduce**

```python
import reflex as rx


class NoCtxState(rx.State):
    clicks: int = 0

    @rx.var(cache=False)
    def heartbeat(self) -> str:
        import datetime
        return datetime.datetime.now().isoformat()

    @rx.event(background=True)
    async def bg_ok(self):
        # never enters `async with self`, returns normally
        pass

    @rx.event(background=True)
    async def bg_raises(self):
        # never enters `async with self`, raises (an illegal write also lands here:
        # self.clicks = 1 raises ImmutableStateError outside the context manager)
        raise RuntimeError("boom")


def index():
    return rx.vstack(
        rx.text(NoCtxState.heartbeat, id="heartbeat"),
        rx.button("ok", on_click=NoCtxState.bg_ok, id="ok"),
        rx.button("raise", on_click=NoCtxState.bg_raises, id="raise"),
    )


app = rx.App()
app.add_page(index)
```

Run the app, watch `#heartbeat`:

- Click `#ok`: exactly one heartbeat DOM update per click (the compatibility flush) — verified in dev and prod.
- Click `#raise`: server logs `[Reflex Backend Exception]` with the traceback, but no delta is emitted — `#heartbeat` never updates.

**Observed behavior**

No compatibility flush when the handler raises; the exception propagates before the flush (same as pre-fix behavior).

**Expected behavior**

The compatibility flush should run even when the handler raises (try/finally around the handler await), so uncached computed vars and preamble-dirty vars stay consistent regardless of how the background task ended.

**Specifics (please complete the following information):**

- Python Version: 3.11
- Reflex Version: 0.9.9a1
- OS: linux
- Browser (Optional): Chromium

**Additional context**

- Severity: low, edge case. Not a regression — matches pre-fix behavior; noting it as a follow-up gap in the flush semantics.
- Found during 0.9.9a1 pre-release testing of the background-delta race fix; the `/noctx` test page and drivers are on branch `claude/reflex-prerelease-testing-t0sd90` under `prerelease-testing/2026-08-27-v0.9.9a1/state_concurrency/`.

### PR DESCRIPTION
**Pull request by FarhanAliRaza**
A background handler that returns without entering async-with-self gets one compatibility delta flush under the state lock (refreshing uncached computed vars and preamble-dirty vars like router_data), but a handler that raises propagated the exception before the flush and the client got no delta at all.

Wrap the handler await in try/finally so the locked flush runs however the task ended; the exception still propagates to the backend exception handler afterwards. A handler that raises after entering the context is unchanged: its context exit already flushed, and the entered flag skips the fallback.

Fixes #6982

### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/reflex-base/src/reflex_base/event/processor/base_state_processor.py [hunk 1]
```diff
@@ -459,29 +459,33 @@ async def _execute_event(
...
-                )
+        try:
+            await process_event(
+                handler=registered_handler.handler,
+                state=proxy,
+                payload=event.payload,
+                root_state=None,
+            )
+        finally:
+            if not proxy._self_entered_context:
```
[the remark this answers is not in the document]
Comment (FarhanAliRaza) [reply packages/reflex-base/src/reflex_base/event/processor/base_state_processor.py:474]: Valid — fixed in 1def528e73d36d8ac7c3c5181f93e0e8eb6454e2. The flush now runs under its own try: when the handler already raised, a flush failure is logged (`logger.exception`) and the handler's exception propagates to `_handle_backend_exception` unchanged; a flush error after a successful handler still raises as before. Regression test: `test_background_flush_failure_does_not_mask_handler_exception` (reproduced the masking first — the backend handler received the flush `ValueError` instead of the handler's `RuntimeError` — and now passes).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/reflex-base/src/reflex_base/event/processor/base_state_processor.py [hunk 2]
```diff
@@ -459,29 +459,33 @@ async def _execute_event(
...
+                    BaseStateToken(
+                        ident=ctx.token,
+                        cls=registered_handler.states[0],
+                    ),
+                    event=event,
+                ) as flush_state:
+                    await chain_updates(
+                        None,
+                        root_state=flush_state._get_root_state(),
+                        handler_name=registered_handler.handler.fn.__qualname__,
```
[the remark this answers is not in the document]
Comment (FarhanAliRaza) [reply packages/reflex-base/src/reflex_base/event/processor/base_state_processor.py]: Valid — fixed in 1def528e73d36d8ac7c3c5181f93e0e8eb6454e2, along the lines suggested: the handler's exception is captured, a flush failure while it is propagating is logged instead of raised, and normal flush errors on successful handlers still propagate. Regression test: `test_background_flush_failure_does_not_mask_handler_exception`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by masenf — APPROVED** [review]
